### ISSUE TITLE
livy_config jars not updating

### ISSUE THREAD

**Issue by pmeakins** [sparklyr/sparklyr#2722]
Hi, 
Using 'R'

config <- livy_config(
    proxy_user = "TESTXXX",
     jars = "http://someurl.com/sparklyr-3.0-2.12.jar"
)

sc <- spark_connect(  master = "REMOTE SERVER 8998", method="livy", config = config )

When I execute the proxy_user is set, but the jars is excluded and defaults to the latest version of sparklyr 1.4, spent two days on this, problem is livy sitting behind proxy, and no whitelist for github by design., just dont seem to be able to pass proxy setting, http_proxy, via httr or even .curlrc/.wgetrc

livy.log

INFO InteractiveSession$: Creating Interactive session 8: [owner: null, request: [kind: spark, proxyUser: Some(TESTXXX), jars: https://github.com/sparklyr/sparklyr/blob/feature/sparklyr-1.4.0/inst/java/sparklyr-3.0-2.12.jar?raw=true, heartbeatTimeoutInSecond: 0]]

**Comment by yitao-li** [comment]
@pmeakins Try the following instead:

```
config <- livy_config(proxy_user = "TESTXXX")
config$sparklyr.livy.jar <-  "http://someurl.com/sparklyr-3.0-2.12.jar"
```

Also, see https://github.com/sparklyr/sparklyr/blob/master/R/config_settings.R

**Comment by pmeakins** [comment]
Hi Yitao-li,  that was a big help thanks,  however proxy settings are ignored, i user httr to also try that way around nothing.  so in the end used local http server to serve the jar - and now working.

Very odd that it didnt pick up the fact that sparklyr was already loaded from adding to the jars dir in the livy install dir.

**Comment by yitao-li** [comment]
@pmeakins So if I understood correctly, the httr proxy settings are ignored when downloading jar files from `http://someurl.com/sparklyr-3.0-2.12.jar`, correct?
I think that is likely something we haven't considered in current implementation of livy connection in sparklyr, and it should be addressed.

Meanwhile you can also consider a proxy setting that applies to the entire R session as a workaround (see https://support.rstudio.com/hc/en-us/articles/200488488-Configuring-R-to-Use-an-HTTP-or-HTTPS-Proxy) if that's something that makes sense in your use case.

**Comment by yitao-li** [comment]
I was thinking configuring the HTTP(s) outbound proxy for Livy server might be possible through the `livy.conf` file, but I don't see any proxy-related options in https://github.com/apache/incubator-livy/blob/master/conf/livy.conf.template . So I don't think the proxy use case you envisioned is fully supported by Livy yet.

So you might have to configure some system-wide HTTP(s) proxy on the Livy server host, or just use the local HTTP(s) workaround you already implemented.

**Comment by pmeakins** [comment]
I am working on connecting SageMaker to EMR, however theres a Proxy whitelist in the way for all outbound, I noticed that livy would alway call out to Github.com to access the sparklyr, I actually in the end added github to the whitelist, but nothing working.

Creating Interactive session 0: [owner: null, request: [kind: spark, proxyUser: None, jars:  https://github.com/sparklyr/sparklyr/blob/feature/sparklyr-1.3.0/inst/java/sparklyr-3.0-2.12.jar?raw=true, heartbeatTimeoutInSecond: 0]]

I found httr and set the proxy that way, also found the livy_config also supported Proxyurl and proxyport but doesnt actually do anything.  So was unable to encourage Livy to use the proxy, I even added .curlrc/.wgetrc with the proxy just in case there was a crude method like curl/wget to download.  I spoofed the github address with Apache and that showed the request was active and no just text, obviously that request failed - but proved the outbound request.

So in the end with the sparklyr.livy.jar I just added an Apache server and located that in the DocumentRoot, and after serving - it all sprung into life,   Fortunately EMR provides a tomcat instance to will piggy back on that to locate the 'jar'

So would be helpful if there was a method to instruct spark_connect / livy_connection to use either systemwide proxy settings or to pass the params.

We are working now, so happy days.  Thanks for the advise, working a treat.
